### DISCUSSION TITLE
HEIC to TIF

### DISCUSSION THREAD

**Discussion by Milk-Dream-sRGB** [ImageMagick/ImageMagick#3080] [Help]
Hello!

ImageMagick 7.0.10-56 Q16 x64 // Windows 10
I tried convert heic to tiff, but resulted is corrupted file.
I tried convert same heic to jpg, then jpg convert to tiff and resulted corrupted file.
I tried convert different image. But, for experement, I convert random jpeg from web to tiff and all goes right.

If I wrong somewhere, please help :(

If someone can help me, can you please tell me if it is possible to do this: convert the heic to tiff and then add the depth of field mask to the alpha channel with one command? Thank you!

[Simple image](https://www.dropbox.com/s/ipuu8z20s1nlvu5/IMG_4022-d.HEIC?dl=0)

**Comment by snibgo** [comment]
Why didn't you show the command you used?

With IM v7.0.8-64 on Windows 8.1, this works fine for me:
```
magick IMG_4022-d.HEIC -colorspace sRGB x.tiff
```

IM can't read a depth of feld mask.

**Reply by Milk-Dream-sRGB** [reply]
Thank you for answer! I'm sorry, I'm new in here and command-line world :)

I tried same command like usually:
`magick IMG_4022-d.HEIC IMG_4022-d.tiff` 
because not see any other way :)

Yes, thanks, your command work fine! it turns out for tiff necessary add specifying parameters?

Latest IM can extract depth mask from .heic shooted in portrait mode! If simple convert heic to jpg result 2 images- one general image and second depth map. When I tried your command result is one tiff with two pages, but Photoshop can open and see only general image :(

**Reply by Milk-Dream-sRGB** [reply]
and problem still have- why I can convert another jpg to tiff without specifying parameters, but when jpg converted from .heic- can't?

**Comment by snibgo** [comment]
The problem is that the main image is encoded as YCbCr (according to both ImageMagick and exiftool). But it has an embedded Apple "Display P3" profile. The profile converts between RGB and XYZ channels. I'm not sure how to most accurately convert to sRGB.

But this may be a more acurate conversion to sRGB:
```
magick IMG_4022-d.HEIC +write x.icc -colorspace sRGB +profile * -profile x.icc -profile sRGB.icc x2.tiff
```
Note that x2.tiff has more saturated colours than x.tiff from my command above.

My old version of IM can't see the depth map. To get just the main image, you probably need [0] like this:
```
magick IMG_4022-d.HEIC[0] -colorspace sRGB x.tiff
```

**Reply by Milk-Dream-sRGB** [reply]
Thank you! I understand!
Yes, right, I see [0] and [1] on "info:" :)
How you think, this code can combine in one line? :)

`magick *-d*.HEIC[0] -set filename:f %t -colorspace sRGB -compress zip %[filename:f]_depth_map-.tif`
`magick *-d*.HEIC[1] -resize 525% -set filename:f %t -colorspace sRGB -compress zip %[filename:f]_depth_map+.tif`

and then I need to combine results into one .tif, but I need HEIC[1] move to alpha in new tiff :)

**Reply by Milk-Dream-sRGB** [reply]
I think it very strange. If simple convert tif with code -colorspace sRGB result is unttaget RGB profile.
If convert with you code with correcting all "x.icc" to "sRGB.icc" result is tif with right color profile, but I can't inderstand how is it. And for what autimatically creat new icc file near photo what I work with... :(

How just create converted HEIC with all metadata (include embeded icc) to tif? :)

**Reply by Milk-Dream-sRGB** [reply]
I try to simple convert .heic to .jpg and all metadata with color _profile_ has stay... How do the same with tif? :)))

**Comment by snibgo** [comment]
I think this should work. For legibility, I have split it into lines.
```
magick ^
  *-d*.HEIC[0] -set filename:f %t -colorspace sRGB -compress zip ^
  %[filename:f]_depth_map-.tif ^
  -delete 0--1 ^
  *-d*.HEIC[1] -resize 525% -set filename:f %t -colorspace sRGB -compress zip ^
  %[filename:f]_depth_map+.tif
```

**Reply by Milk-Dream-sRGB** [reply]
Thank you very much!
But CMD show error :(

**Comment by snibgo** [comment]
Yeah, sorry, I missed out a "-write", like this:
```
magick ^
  *-d*.HEIC[0] -set filename:f %t -colorspace sRGB -compress zip ^
  -write %[filename:f]_depth_map-.tif ^
  -delete 0--1 ^
  *-d*.HEIC[1] -resize 525% -set filename:f %t -colorspace sRGB -compress zip ^
  %[filename:f]_depth_map+.tif
```

**Reply by Milk-Dream-sRGB** [reply]
Thank you!!! It's works!!!
Tell me please, what the it doing:
`-delete 0--1` ?

**Reply by Milk-Dream-sRGB** [reply]
Really thank you! I dreamed about it few years 😅
And, I think last question...
I found command:
`magick *.tif *.png -alpha off -compose CopyOpacity -composite *.tif`
which make what I need to do, but it applying "opacity mask", but I need just a copy to alpha channel. How you think- it can be?..

at now I have this command:
`magick *-d*.HEIC[0] -set filename:f %t -colorspace sRGB -write 1.tif -delete 0--1 *-d*.HEIC[1] -resize 525% -colorspace sRGB 2.png`
And I wanna finish it with combine resulted images and give new image same name from original :)
